R/salesforce.R

```diff
@@ -0,0 +1,97 @@
+#' API to login to Salesforce
+#' export
+#' @param sever - (optional) login server
+#' @param username - Salesforce login user name
+#' @param password - Salesforce login password
+#' @param securityToken - (optional) security token if required
+#' This API
+loginToSalesforce <- function(server = NULL, username, password, securityToken = NULL){
+  if (!requireNamespace("salesforcer")) {
+    stop("package salesforcer must be installed.")
+  }
+  if (is.null(server)) { # if login server was not provided, try it with the default login server.
+    server = "https://login.salesforce.com"
+  }
+  if (is.null(securityToken)) {
+    salesforcer::sf_auth(login_url = server, username = username, password = password, cache = FALSE)
+  } else {
+    salesforcer::sf_auth(login_url = server, username = username, password = password, security_token = securityToken, cache = FALSE)
+  }
+}
+
+#' API to get list of Standard and Custom Objects from Salesforce
+#' export
+#' @param sever - (optional) login server
+#' @param username - Salesforce login user name
+#' @param password - Salesforce login password
+#' @param securityToken - (optional) security token if required
+querySaleseforceMetadata <- function(server = NULL, username, password, securityToken = NULL){
+  if (!requireNamespace("salesforcer")) {
+    stop("package salesforcer must be installed.")
+  }
+  loginToSalesforce(server = server, username = username, password = password, securityToken = securityToken)
+  my_queries <- list(list(type='CustomObject'))
```

Comment:
Does this query include query for standard objects?

Reply:
Added comment.